src/Catel.Core/Catel.Core.Shared/Services/Interfaces/IObjectIdGenerator.cs

```diff
@@ -0,0 +1,41 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="IObjectIdGenerator.cs" company="Catel development team">
+//   Copyright (c) 2008 - 2018 Catel development team. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Catel.Services
+{
+    /// <summary>
+    /// The object id generator service.
+    /// </summary>
+    /// <typeparam name="TUniqueIdentifier">The unique identifier type</typeparam>
+    public interface IObjectIdGenerator<TUniqueIdentifier>
+    {
+        /// <summary>
+        /// Gets the unique identifier for the specified type.
+        /// </summary>
+        /// <returns>A new unique identifier.</returns>
+        TUniqueIdentifier GetUniqueIdentifier();
+
+        /// <summary>
+        /// Gets the unique identifier for the specified type.
+        /// </summary>
+        /// <returns>A new unique identifier but if <paramref name="reuse"/> is <c>true</c> will return a released identifier.</returns>
+        TUniqueIdentifier GetUniqueIdentifier(bool reuse);
```

Comment:
Since this is a new interface, we could just introduce this as an optional parameter (no breaking change here). Because what's the value used when using `GetUniqueIdentifier()`, it's not explained in the docs.

Reply:
Fixed